src/index.cpp

```diff
@@ -1192,9 +1192,19 @@ void Index<T, TagT, LabelT>::search_for_point_and_prune(int location, uint32_t L
     }
 
     prune_neighbors(location, pool, pruned_list, scratch);
-
-    assert(!pruned_list.empty());
-    assert(_final_graph.size() == _max_points + _num_frozen_pts);
+    // skip full dupl vectors
+    NeighborPriorityQueue &best_L_nodes = scratch->best_l_nodes();
+    _full_duplicated =
+        best_L_nodes.size() > 0 && best_L_nodes.operator[](0).distance == 0 && best_L_nodes.operator[](0).expanded;
```

Comment:
If I understand the pruning right, it would be better to prune to the max number of neighbors by some stable way like vector ID, otherwise you can have more than the legal number of neighbors